network_models/experiment.py

```diff
@@ -78,14 +78,41 @@ def experiment(network_model, output_path, dataset_model=None, args=None, config
         os.makedirs(model_output_path)
     model_checkpoint = ModelCheckpoint(model_output_path)
 
-    history = network_model.fit_generator(dataset_model.get_training_generator(),
+    # Default to the simple generator unless the `training_multiprocess`
+    # flag is set within the config file
+    if "training_multiprocess" in config and config["training_multiprocess"]:
+        print("Training will run in **multiple** processes. Change your config if this is undesired.")
+        training_generator = dataset_model.get_training_generator_multiprocess()
+        use_multiprocessing = True
+    else:
+        print("Training will run in a **single** processes. Change your config if this is undesired.")
+        training_generator = dataset_model.get_training_generator()
+        use_multiprocessing = False
+
+    # Default to getting all the training data unless the
+    # `validation_generator` flag is set within the config file
+    if "validation_generator" in config and config["validation_generator"]:
+        validation_generator = dataset_model.get_validation_generator()
+    else:
+        print("All validation data will be loaded into memory. Change your config if this is undesired.")
+        validation_generator = dataset_model.get_validation_data()
+
+    # Workers specify the number of processes and/or threads executing
+    if "workers" in config:
```

Comment:
After our conversation I feel that this line should be like this:

if "workers" in config and "training_multiprocess" in config and config["training_multiprocess"]:

That way we avoid a crash when workers is greater than 1 but there is no training_multiprocess.  Give it a try.

Reply:
How about I put a warning and an exit in the event it will attempt to do multi-threaded instead of multi-process? Then the behavior matches what is expected via the config-- it just refuses to execute it.